src/Components/MemberLogged/SubscriptionsMember/index.jsx

```diff
@@ -0,0 +1,131 @@
+
+  useEffect(() => {
+    getSubscriptions(dispatch);
+    getActivities(dispatch);
+  }, [dispatch]);
+
+  useEffect(() => {
+    getActivitiesName().then((data) => {
+      setSubscription(data);
+    });
+  }, [subscriptions, activities]);
+
+  const getActivitiesName = async () => {
+    let arraySubs = [];
+    const filterSubscription = subscriptions.filter((item) => item.members._id === member._id);
+
+    await filterSubscription.forEach((sub) => {
+      activities?.forEach((act) => {
+        if (sub.classes.activity === act._id) {
+          arraySubs.push({
+            subId: sub._id,
+            activityName: act.name,
+            day: sub.classes.day,
+            time: sub.classes.time
+          });
+        }
+      });
+    });
+    return arraySubs;
+  };
+
+  const handleDeleteActivity = () => {
+    dispatch(deleteSubscription(idToDelete));
+    setShowConfirmModal(false);
+  };
+
+  const handleDeleteButton = (id) => {
+    setShowConfirmModal(true);
+    setIdToDelete(id);
+  };
+
+  return (
+    <>
+      {pending && (
+        <div className={styles.loader}>
+          <Loader />
+        </div>
+      )}
+      {!pending && (
+        <div className={styles.container}>
+          <h2 className={styles.title}>Your subscriptions</h2>
+          {subscription.length !== 0 ? (
+            <table className={styles.table}>
+              <thead className={styles.thead}>
+                <tr>
+                  <th className={styles.th}>Activity</th>
+                  <th className={styles.th}>Day</th>
+                  <th className={styles.th}>Time</th>
+                  <th className={styles.th}></th>
+                </tr>
+              </thead>
+              <tbody>
+                {subscription?.map((item) => {
+                  return (
+                    <tr className={styles.tr} key={item.subId}>
+                      <td className={styles.td}>{item.activityName}</td>
+                      <td className={styles.td}>{item.day}</td>
+                      <td className={styles.td}>{item.time}</td>
+                      <td className={styles.button}>
+                        <div className={styles.buttonContainer}>
+                          <Button
+                            text="Unsubscribe"
+                            classNameButton="deleteButton"
+                            action={() => handleDeleteButton(item.subId)}
+                          />
+                        </div>
+                      </td>
+                    </tr>
+                  );
+                })}
+              </tbody>
+            </table>
+          ) : (
+            <p className={styles.text}>You dont have any subscription</p>
+          )}
+        </div>
+      )}
+      {showConfirmModal && (
+        <ConfirmModal
+          handler={() => setShowConfirmModal(false)}
+          title="Unsubscribe Activity"
+          reason="delete"
+          onAction={() => handleDeleteActivity()}

```

Comment:
In the "onAction" attribute of "ConfirmModal", instead of using an arrow function, it would be simpler and clearer to directly pass the reference of the "handleDeleteActivity" function. This will simplify the code and maintain the desired functionality.
Using an anonimous arrow function in this case can create some unnecessary overhead. By directly passing the function reference, we avoid creating an additional function and improve code readability.
Unlike the case in line 96 where you need to pass an argument, which requires using an arrow function.
```suggestion
          onAction={handleDeleteActivity}

```